odl/util/utility.py

```diff
@@ -147,3 +129,154 @@ def __new__(cls, name, this_bases, d):
                 return type.__new__(cls, name, (), d)
             return meta(name, bases, d)
     return metaclass('temporary_class', None, {})
+
+
+def is_scalar_dtype(dtype):
+    """Whether a datatype is scalar or not."""
+    return np.issubsctype(dtype, np.number)
+
+
+def is_int_dtype(dtype):
+    """Whether a datatype is integer or not."""
+    return np.issubsctype(dtype, np.integer)
+
+
+def is_real_dtype(dtype):
+    """Whether a datatype is real (including integer) or not."""
+    return is_scalar_dtype(dtype) and not is_complex_floating_dtype(dtype)
+
+
+def is_real_floating_dtype(dtype):
+    """Whether a NumPy datatype is real complex-floating or not."""
+    return np.issubsctype(dtype, np.floating)
+
+
+def is_complex_floating_dtype(dtype):
+    """Whether a NumPy datatype is complex floating-point or not."""
+    return np.issubsctype(dtype, np.complexfloating)
+
+
+def preload_call_with(instance, mode):
+    """Decorator to preload the first argument of a call method.
```

Comment:
Here it means that the first argument of the function is "preloaded" with `instance`. I'm happy to take suggestions for better names.